### PR TITLE
OCPBUGS-32377: [IBU] Remove SR-IOV workaround from upgrade handler

### PR DESCRIPTION
**Pull request by leo8a**
Removes the SR-IOV workaround from the upgrade handler.

I've kept the support for gathering info about SR-IOV operator in the must-gather thou, if this needs to be also removed, just let me know.

/cc @donpenney @browsell

### CODE REVIEW

File: internal/healthcheck/healthcheck.go [hunk 1]
```diff
@@ -82,12 +80,6 @@ func HealthChecks(ctx context.Context, c client.Reader, l logr.Logger) error {
 	}
 
 	if clusterOperatorsReady && clusterServiceVersionsReady {
-		// Only check SriovNetworkNodeState once cluster operators and CSVs are stable
-		if err := IsSriovNetworkNodeReady(ctx, c, l); err != nil {
```
Comment (donpenney) [inline internal/healthcheck/healthcheck.go:86]: We still need the SRIOV network node ready health check. It's just the workaround in the upgrade_handlers.go that can be removed

**Review by donpenney — CHANGES_REQUESTED** [review]

**Comment by donpenney** [comment]
The only functionality to be removed here is the sriovNetworkNodeStateWorkaround function. The health check is still required

File: internal/healthcheck/healthcheck.go (hunk 1 above)
Comment (leo8a) [reply internal/healthcheck/healthcheck.go]: oh gotcha, I've shrunk the scope of the PR

**Review by donpenney — COMMENTED** [review]
/lgtm

**Pull request merged**